markdown.dtx

```diff
@@ -31016,17 +31057,40 @@ M.extensions.fenced_divs = function(blank_before_div_fence)
 %
 % If the `blank_before_div_fence` parameter is `false`, we will have the
 % closing div at the beginning of a line break the current paragraph if
-% we are currently nested in a div.
+% we are currently nested in a div and the indentat matches the opening
+% div fence.
 %
 % \end{markdown}
 %  \begin{macrocode}
-      local function check_div_level(s, i, current_level) -- luacheck: ignore s i
-        current_level = tonumber(current_level)
-        return current_level > 0
+      local function is_inside_div()
+        local function check_div_level(s, i, fenced_div_level) -- luacheck: ignore s i
+          fenced_div_level = tonumber(fenced_div_level)
+          return fenced_div_level > 0
+        end
+
+        return Cmt(Cb("fenced_div_level"), check_div_level)
+      end
+
+      local function check_indent()
+        local function compare_indent(s, i, indent_table, -- luacheck: ignore s i
+                                      fenced_div_num_opening_indents, fenced_div_level)
+          fenced_div_level = tonumber(fenced_div_level)
+          local num_current_indents = 0
+                    if indent_table.indents ~= nil then
+            num_current_indents = #indent_table.indents
+          end
+          local num_opening_indents = fenced_div_num_opening_indents[fenced_div_level]
+          return num_current_indents == num_opening_indents or #indent_table.continues == num_opening_indents
```

Comment:
As discussed in https://github.com/Witiko/markdown/pull/431/files#r1550351531:

```suggestion
          return num_current_indents == num_opening_indents or #indent_table.curr_indent == num_opening_indents
```

(Perhaps `indent_table.curr_indents` or `indent_table.curr_line_indents` are more self-documenting and more consistent with `indent_table.indents`?)

Reply:
Furthermore, do we need both `#indent_table.continues` and `num_current_indents` (i.e. `#indent_table.indents`) here? It seems that having `num_current_indents = (indent_table.continues ~= nil and #indent_table.continues) or (indent_table.indents ~= nil and #indent_table.indents) or 0` would be simpler and allow us to say simply `return num_current_indents == num_opening_indents`. If `indent_table.continues ~= nil` whenever `indent_table.indents ~= nil`, then we can simplify and have just `num_current_indents = (indent_table.continues ~= nil and #indent_table.continues) or 0`.